### PR TITLE
mcs, schedule: use txn in label rule manager

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Ref #5839

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/labeler/labeler.go [hunk 1]
```diff
@@ -88,11 +89,10 @@ func (l *RegionLabeler) checkAndClearExpiredLabels() {
 			continue
 		}
 		if len(rule.Labels) == 0 {
-			err = l.storage.DeleteRegionRule(key)
-			delete(l.labelRules, key)
+			err = l.DeleteLabelRuleLocked(key)
 			deleted = true
```
Comment (rleungx) [inline pkg/schedule/labeler/labeler.go:93]: If there is an error, `deleted` is still true?

File: pkg/schedule/labeler/labeler.go [hunk 2]
```diff
@@ -88,11 +89,10 @@ func (l *RegionLabeler) checkAndClearExpiredLabels() {
...
 		} else {
-			err = l.storage.SaveRegionRule(key, rule)
+			err = l.SetLabelRuleLocked(rule)
```
Comment (rleungx) [inline pkg/schedule/labeler/labeler.go:95]: I prefer not to change the unnecessary code, it does one more check and memory modification.

File: pkg/schedule/labeler/labeler.go [hunk 3]
```diff
@@ -124,7 +124,7 @@ func (l *RegionLabeler) loadRules() error {
 		return err
 	}
 	for _, d := range toDelete {
-		if err = l.storage.DeleteRegionRule(d); err != nil {
+		if err = l.DeleteLabelRuleLocked(d); err != nil {
```
Comment (rleungx) [inline pkg/schedule/labeler/labeler.go:127]: ditto, previous, it didn't change the memory.

Comment (lhy1024) [reply pkg/schedule/labeler/labeler.go:129]: we should keep storage and memory consistent

Comment (rleungx) [reply pkg/schedule/labeler/labeler.go:129]: The rule won't be add into memory.

**Review by rleungx — APPROVED** [review]

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**